### DISCUSSION TITLE
Inherit from Inline

### DISCUSSION THREAD

**Discussion by christianzech** [AvaloniaUI/Avalonia#9076] [General]
Is it intentional that BuildTextRun() and AppendText() of Inline are internal? This disallows inheriting from Inline and making own inline elements. Atm, I need to inherit from Run instead. But as I want to control the variations of a OpenType font, the internal handling of the Font seems to break my font variation settings (especially for the weight). 

Thanks.

**Comment by Gillibald** [comment]
How are you customizing the Inlines exactly? Are you just altering TextRunProperies or are you adding your own TextRun implementations? We can mark CreateTextRunProperties virtual so this becomes customizable.

**Reply by christianzech** [reply]
Good question, I still try to find out what's the best approach. What would be nice to have from my side is a symbol/icon inline similar to UWPs FontIcon or SymbolIcon (not only as control, but also as inline to be part of a rich text). Therefore I'd use FluentSystemIcons or MaterialIcons font but dont want to always have to set FontFamily and text glyph by hand. As I can't inherit from Inline directly, I inherit from Run and set the FontFamily property. The Text property is set internally to the desired char by setting a Symbol or Glyph property and using lookup tables. But FontFamily and Text can be (accidentially) changed by the user so that you might get unexpected results. That's why I'd like to directly inherit from Inline. But in that case I don't only need access to CreateTextRunProperties() but also to BuildTextRun() and AppendText().
Furthermore it might be interesting to also add an Inline similar to PathIcon. In that case I'd also need an own TextRun implementation or maybe consume DrawableTextRun. 
On the other side, if FontIcon, SymbolIcon and PathIcon were controls, they could be embedded in rich text using InlineUIContainer. But I had concerns regarding performance and correct baseline alignment. 

The other point mentioned is that I was also plaing with MaterialSymbols font. As this is a variable OpenType font, you don't need specific fonts for different styles (filled, outlined,...) so that you dont have to change the FontFamily. HarfBuzz supports OpenType variable axes and I'm able to set the fill axis corretly. But all other axes (weight, optical-size) do not work. I thought it might be a problem as the weight of the typeface is also set using TextElement.FontWeight but it seems that Skia doesn't load the correct SkTypeface from the font file.
